frame/node-authorization/src/lib.rs

```diff
@@ -0,0 +1,352 @@
+	type Event: From<Event> + Into<<Self as frame_system::Trait>::Event>;
+
+    /// The maximum number of authorized nodes that are allowed to set
+    type MaxAuthorizedNodes: Get<u32>;
+
+    /// The node identifier type for the runtime.
+    type NodeId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay
+        + Ord + Default;
+
+    /// The origin which can add a authorized node.
+    type AddOrigin: EnsureOrigin<Self::Origin>;
+
+    /// The origin which can remove a authorized node.
+    type RemoveOrigin: EnsureOrigin<Self::Origin>;
+
+    /// The origin which can swap the authorized nodes.
+    type SwapOrigin: EnsureOrigin<Self::Origin>;
+
+    /// The origin which can reset the authorized nodes.
+    type ResetOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Weight information for extrinsics in this pallet.
+    type WeightInfo: WeightInfo;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as NodeAuthorization {}
+    add_extra_genesis {
+        config(nodes): Vec<T::NodeId>;
+        build(|config: &GenesisConfig<T>| {
+            sp_io::storage::set(well_known_keys::NODE_ALLOW_LIST, &config.nodes.encode());
+        });
+    }
+}
+
+decl_event! {
+    pub enum Event {
+        /// The given node was added; see the transaction for node id.
+        NodeAdded,
+        /// The given node was removed; see the transaction for node id.
+        NodeRemoved,
+        /// Two given nodes were swapped; see the transaction for node id.
+        NodesSwapped,
+        /// The given nodes were reset; see the transaction for new set.
+        NodesReset,
+    }
+}
+
+decl_error! {
+    /// Error for the node authorization module.
+    pub enum Error for Module<T: Trait> {
+        /// Too many authorized nodes.
+        TooManyNodes,
+        /// The node is already joined in the list.
+        AlreadyJoined,
+        /// The node doesn't exist in the list.
+        NotExist,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        /// The maximum number of authorized nodes
+        const MaxAuthorizedNodes: u32 = T::MaxAuthorizedNodes::get();
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Add a node to the allowlist.
+        ///
+        /// May only be called from `T::AddOrigin`.
+        ///
+        /// - `node_id`: identifier of the node, it's likely the public key of ed25519 keypair.
+        #[weight = (T::WeightInfo::add_node(), DispatchClass::Operational)]
+        pub fn add_node(origin, node_id: T::NodeId) {
+            T::AddOrigin::ensure_origin(origin)?;
+
+            let mut nodes = Self::get_allowlist();
+            ensure!(nodes.len() < T::MaxAuthorizedNodes::get() as usize, Error::<T>::TooManyNodes);
+
+            let location = nodes.binary_search(&node_id).err().ok_or(Error::<T>::AlreadyJoined)?;
+            nodes.insert(location, node_id);
+            Self::put_allowlist(nodes);
```

Comment:
`allowlist` should be snake case `allow_list` everywhere. This improves readability in the rust code and from the JS API.